docs/source/daml/reference/interfaces.rst

```diff
@@ -0,0 +1,52 @@
+.. Copyright (c) 2022 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+.. SPDX-License-Identifier: Apache-2.0
+
+.. _interfaces:
+
+Reference: Interfaces
+#####################
+
+In Daml, an interface defines an abstract type which specifies the behavior
+that a template must implement. This allows decoupling such behavior from its
+implementation, so other developers can write applications in terms of the
+interface instead of the concrete template.
+
+Interface declaration
+*********************
+
+An interface declaration is somewhat similar to a template declaration.
+
+Interface name
+--------------
+
+.. literalinclude:: ../code-snippets-dev/Interfaces.daml
+   :language: daml
+   :start-after: -- INTERFACE_NAME_BEGIN
+   :end-before: -- INTERFACE_NAME_END
+
+- This is the name of the interface.
+- It's preceded by the keyword ``interface`` and followed by the keyword ``where``.
+- It must begin with a capital letter, like any other type name.
+
+Interface methods
+-----------------
+
+.. literalinclude:: ../code-snippets-dev/Interfaces.daml
+   :language: daml
+   :start-after: -- INTERFACE_METHODS_BEGIN
+   :end-before: -- INTERFACE_METHODS_END
+
+- An interface may define any number of methods
+- Methods are visible as functions at the top level and in interface choices
+  (see :ref:`interface-choices` below).
+- The type of a method will always be prefixed with
```

Comment:
there must be a less convoluted way to say this but I can't think of one right now